### PR TITLE
Implement Color Operations for `Color`

### LINKED ISSUE #13214: Implement all non-trivial color traits on `Color` type
## What problem does this solve or what need does it fill?

If #12212 is closed as won't fix, the developer experience of working with our assorted fields is much worse than it needs to be.

Rather than simply being able to call:

```rust
*color = color.lighten(0.2)
```

they must write

```rust
let hsla = Hsla::from(color);
let light_hsla = hsla.lighten(0.2);
*color = Color::from(light_hsla);
```

## What solution would you like?

If an operation is supported in the supplied color space, call the corresponding method.

If it is not, convert to the most reasonable color space for that operation (usually Oklaba), perform the operation, and then convert back.

## What alternative(s) have you considered?

We could choose a single opinionated perceptual color space for working with all parts of UI code.

This is limiting, and particularly painful since it would force const definitions for palettes to be done directly in that color space. This isn't feasible because of limitations on const float arithmetic and const traits.

### PR DESCRIPTION
**Pull request by bushrat011899**
# Objective

- Fixes #13214

## Solution

Delegates to internal type when possible, otherwise uses `ChosenColorSpace` as an intermediary. This _will_ double convert, but this is considered an acceptable compromise since use of specific colour types in performance critical colour operations is already encouraged.

`ChosenColorSpace` is `Oklcha` since it's perceptually uniform while supporting all required operations, and in my opinion is the "best" for this task. Using different spaces for different operations will make documenting this double-conversion behaviour more challenging.

## Testing

Changes straightforward enough to not require testing beyond current CI in my opinion.

---

## Changelog

- Implemented the following traits for `Color`:
  - `Luminance`
  - `Hue`
  - `Mix`
  - `EuclideanDistance`
  - `ClampColor`
- Added documentation to `Color` explaining the behaviour of these operations (possible conversion, etc.)

### CODE REVIEW

File: crates/bevy_color/src/color.rs [hunk 1]
```diff
@@ -11,6 +12,31 @@ use bevy_reflect::prelude::*;
 /// <div>
 #[doc = include_str!("../docs/diagrams/model_graph.svg")]
 /// </div>
+///
+/// # Operations
+///
+/// [`Color`] supports all the standard color operations, such as [mixing](Mix),
+/// [luminance](Luminance) and [hue](Hue) adjustment, [clamping](ClampColor),
+/// and [diffing](EuclideanDistance). These operations delegate to the concrete color space contained
+/// by [`Color`], but will convert to [`Oklch`](Oklcha) for operations which aren't supported in the
```
Comment (tychedelia) [inline crates/bevy_color/src/color.rs:21]: Maybe add a tiny comment about the rationale for `Oklcha`?

**Review by tychedelia — APPROVED** [review]
Personally, I am starting to really like the enum approach. I think this is going to be very expressive for our users who are able to work with `Color` rather than a specific space.

**Review by alice-i-cecile — APPROVED** [review]
I really like the end results here: it provides a great balance between correctness, performance and usability.

**Review by NthTensor — APPROVED** [review]

**Comment by bushrat011899** [comment]
@alice-i-cecile should be good to go now!

**Pull request merged**